shared_model/interfaces/commands/add_asset_quantity.hpp

```diff
@@ -31,26 +30,37 @@ namespace shared_model {
      * Add amount of asset to an account
      */
     class AddAssetQuantity
-        : public Primitive<AddAssetQuantity, iroha::model::AddAssetQuantity> {
+        : public Hashable<AddAssetQuantity, iroha::model::AddAssetQuantity> {
      public:
       /**
        * @return Identity of user, that add quantity
        */
       virtual const types::AccountIdType &accountId() const = 0;
-
-      /// Type of asset id
-      using AssetIdType = std::string;
       /**
        * @return asset identifier
        */
-      virtual const AssetIdType &assetId() const = 0;
-
-      /// Type of amount
-      using AmountType = iroha::Amount;
+      virtual const types::AssetIdType &assetId() const = 0;
       /**
        * @return quantity of asset for adding
        */
-      virtual const AmountType &amount() const = 0;
+      virtual const types::AmountType &amount() const = 0;
+
+      std::string toString() const override {
+        return detail::PrettyStringBuilder()
+            .init("AddAssetQuantity")
+            .append("account_id", accountId())
+            .append("asset_id", assetId())
+            .append("amount", amount().to_string())
+            .finalize();
+      }
+
+      OldModelType *makeOldModel() const override {
+        auto oldModel = new iroha::model::AddAssetQuantity;
```

Comment:
why do you use raw `new` but not smart pointers?
I bet caller will not know that `makeOldModel` allocates memory, thus he will not delete it.

Reply:
In business logic, you may wrap raw pointer with smart, by the way.
General reason to provide this API is flexible usage. In your business logic, you may wrap it with a smart pointer if it required.

Reply:
Okay. At least [here](https://github.com/hyperledger/iroha/pull/672/files#diff-bb97a105e92729a26663fc51ca864abbR53) and [here](https://github.com/hyperledger/iroha/pull/672/files#diff-92ecbf03cb6870a74c08cb497fdfa38fR56) the pointer is stored inside a structure and not wrapped by a smart pointer. At the same time, destructor does not delete the allocated memory.

Caller will not know that some memory is allocated inside. Therefore, it introduces a memory leak.

Reply:
UPD: whenever `makeOldModel` is called, there should be written a destructor which deallocates memory, otherwise it is a memory leak.

Reply:
UPD: ah, I see that templated makeOldModel does not allocate memory on heap. 

It sounds like a костыль

Reply:
Yes, I agree with your issue, but for this method, our `makeOldModel` API not applied because it requires template parameter for conversion. And this method not designed as heap allocated because for the array it is not required (it is not polymorphic).